ts_src/confidential.ts

```diff
@@ -65,882 +9,406 @@ export interface UnblindOutputResult {
+    nInputs: number;
+    usedInputs: Buffer;
+    data: Buffer;
+  };
+  initialize: (
+    inputTags: Array<Buffer>,
+    inputTagsToUse: number,
+    outputTag: Buffer,
+    maxIterations: number,
+    seed: Buffer,
+  ) => {
+    proof: { nInputs: number; usedInputs: Buffer; data: Buffer };
+    inputIndex: number;
+  };
+  generate: (
+    proof: { nInputs: number; usedInputs: Buffer; data: Buffer },
+    inputTags: Array<Buffer>,
+    outputTag: Buffer,
+    inputIndex: number,
+    inputBlindingKey: Buffer,
+    outputBlindingKey: Buffer,
+  ) => { nInputs: number; usedInputs: Buffer; data: Buffer };
+  verify: (
+    proof: { nInputs: number; usedInputs: Buffer; data: Buffer },
+    inputTags: Array<Buffer>,
+    outputTag: Buffer,
+  ) => boolean;
 }
 
-interface RngOpts {
-  rng?(arg0: number): Buffer;
+export interface ZKP {
+  ecdh: Ecdh;
+  ec: Ec;
+  surjectionproof: SurjectionProof;
+  rangeproof: RangeProof;
+  pedersen: Pedersen;
+  generator: Generator;
 }
 
-export type KeysGenerator = (opts?: RngOpts) => {
-  publicKey: Buffer;
-  privateKey: Buffer;
-};
+export class Confidential {
+  constructor(private zkp: ZKP) {}
 
-export class ZKPValidator implements PsetBlindingValidator {
-  async verifyValueRangeProof(
-    valueCommit: Buffer,
-    assetCommit: Buffer,
-    proof: Buffer,
-    script: Buffer,
-  ): Promise<boolean> {
-    try {
-      return await rangeProofVerify(valueCommit, assetCommit, proof, script);
-    } catch (ignore) {
-      return false;
-    }
+  nonceHash(pubkey: Buffer, privkey: Buffer): Buffer {
+    return crypto.sha256(this.zkp.ecdh(pubkey, privkey));
   }
 
-  async verifyAssetSurjectionProof(
-    inAssets: Buffer[],
-    inAssetBlinders: Buffer[],
-    outAsset: Buffer,
-    outAssetBlinder: Buffer,
-    proof: Buffer,
-  ): Promise<boolean> {
-    try {
-      return await surjectionProofVerify(
-        inAssets,
-        inAssetBlinders,
-        outAsset,
-        outAssetBlinder,
-        proof,
-      );
-    } catch (ignore) {
-      return false;
-    }
-  }
-
-  async verifyBlindValueProof(
-    valueCommit: Buffer,
-    assetCommit: Buffer,
-    proof: Buffer,
-  ): Promise<boolean> {
-    try {
-      return await rangeProofVerify(valueCommit, assetCommit, proof);
-    } catch (ignore) {
-      return false;
-    }
-  }
-
-  async verifyBlindAssetProof(
-    asset: Buffer,
-    assetCommit: Buffer,
-    proof: Buffer,
-  ): Promise<boolean> {
-    try {
-      return await assetBlindProofVerify(asset, assetCommit, proof);
-    } catch (ignore) {
-      return false;
-    }
-  }
-}
-
-export class ZKPGenerator implements PsetBlindingGenerator {
-  static fromOwnedInputs(ownedInputs: OwnedInput[]): ZKPGenerator {
-    const bg = new ZKPGenerator();
-    bg.ownedInputs = ownedInputs;
-    return bg;
+  valueBlindingFactor(
+    inValues: string[],
+    outValues: string[],
+    inGenerators: Buffer[],
+    outGenerators: Buffer[],
+    inFactors: Buffer[],
+    outFactors: Buffer[],
```

Comment:
maybe better to use more familiar naming?

```suggestion
    inAssets: Buffer[],
    outAssets: Buffer[],
    inBlinders: Buffer[],
    outBlinders: Buffer[],
```